axiom/tests/test_span_event.c

```diff
@@ -434,6 +475,58 @@ static void test_span_events_extern_get_and_set(void) {
   nr_span_event_destroy(&span);
 }
 
+static void test_span_event_message_string_get_and_set(void) {
+  nr_span_event_t* event = nr_span_event_create();
+
+  // Test : that is does not blow up when we give the setter a NULL pointer
+  nr_span_event_set_message(NULL, NR_SPAN_MESSAGE_DESTINATION_NAME, "wallaby");
+  tlib_pass_if_null(
+      "the destination name should still be NULL",
+      nr_span_event_get_message(event, NR_SPAN_MESSAGE_DESTINATION_NAME));
+  nr_span_event_set_message(event, NR_SPAN_MESSAGE_DESTINATION_NAME, NULL);
+  tlib_pass_if_null(
+      "given a NULL value we should get a NULL",
+      nr_span_event_get_message(event, NR_SPAN_MESSAGE_DESTINATION_NAME));
+
+  // Test : the getter should not blow up when we send it an event with a NULL
+  // component
+  tlib_pass_if_null(
+      "NULL event -> NULL component",
+      nr_span_event_get_message(NULL, NR_SPAN_MESSAGE_DESTINATION_NAME));
+
+  // Test : setting the destination name back and forth behaves as expected
+  nr_span_event_set_message(event, NR_SPAN_MESSAGE_DESTINATION_NAME, "chicken");
+  tlib_pass_if_str_equal(
+      "should be the component we set 1", "chicken",
+      nr_span_event_get_message(event, NR_SPAN_MESSAGE_DESTINATION_NAME));
+  nr_span_event_set_message(event, NR_SPAN_MESSAGE_DESTINATION_NAME, "oracle");
+  tlib_pass_if_str_equal(
+      "should be the component we set 2", "oracle",
+      nr_span_event_get_message(event, NR_SPAN_MESSAGE_DESTINATION_NAME));
+
+  // Test : setting the messaging system back and forth behaves as expected
+  nr_span_event_set_message(event, NR_SPAN_MESSAGE_MESSAGING_SYSTEM, "chicken");
+  tlib_pass_if_str_equal(
+      "should be the component we set 1", "chicken",
+      nr_span_event_get_message(event, NR_SPAN_MESSAGE_MESSAGING_SYSTEM));
+  nr_span_event_set_message(event, NR_SPAN_MESSAGE_MESSAGING_SYSTEM, "oracle");
+  tlib_pass_if_str_equal(
+      "should be the component we set 2", "oracle",
+      nr_span_event_get_message(event, NR_SPAN_MESSAGE_MESSAGING_SYSTEM));
+
+  // Test : setting the server address back and forth behaves as expected
+  nr_span_event_set_message(event, NR_SPAN_MESSAGE_SERVER_ADDRESS, "chicken");
+  tlib_pass_if_str_equal(
+      "should be the component we set 1", "chicken",
+      nr_span_event_get_message(event, NR_SPAN_MESSAGE_SERVER_ADDRESS));
+  nr_span_event_set_message(event, NR_SPAN_MESSAGE_SERVER_ADDRESS, "oracle");
+  tlib_pass_if_str_equal(
+      "should be the component we set 2", "oracle",
+      nr_span_event_get_message(event, NR_SPAN_MESSAGE_SERVER_ADDRESS));
+
```

Comment:
Do we care to test with invalid input, i.e. member argument passed is out of valid range like this: nr_span_event_get_message(event, 255);?

Reply:
Please see: https://github.com/newrelic/newrelic-php-agent/pull/991/commits/862604b081029206c2072275588076691d30560a